gslib/utils/text_util.py

```diff
@@ -148,10 +149,7 @@ def DecodeLongAsString(long_to_convert):
   Returns:
     String decoded from the input long.
   """
-  if isinstance(long_to_convert, basestring):
-    # Already converted.
-    return long_to_convert
-  return hex(long_to_convert)[2:-1].decode('hex')
+  return str(long_to_convert)
```

Comment:
This doesn't give us the same result we had before.  Let's take a real x-amz-version-id value and test it:

'9PpsRjBGjBh90IvIS96dgRc_UL6NyGqD'
(encoded long: 25923956239092482442895228561437790190304192615858167521375267910356975448388L)

(Note that all the `long`s we're dealing with were originally strings from header values, and thus should be convertible back to utf-8 -- probably ascii if AWS was strict about those things, but I'm not sure how much I trust the strict ascii-ness of their values)

Py2:
```
>>> import binascii
>>> var = 25923956239092482442895228561437790190304192615858167521375267910356975448388L
>>> hex(var)
'0x39507073526a42476a42683930497649533936646752635f554c364e79477144L'

# Just strips off the "0x" and trailing "L". Note that the L isn't there in Python 3 though...
>>> hex(var)[2:-1]
'39507073526a42476a42683930497649533936646752635f554c364e79477144'

>>> hex(var)[2:-1].decode('hex')
'9PpsRjBGjBh90IvIS96dgRc_UL6NyGqD'

# Py2 just uses this under the hood for decode('hex'):
>>> binascii.unhexlify(hex(var)[2:-1])
'9PpsRjBGjBh90IvIS96dgRc_UL6NyGqD'
```

But this function gives us something different:
```
>>> str(var)
'25923956239092482442895228561437790190304192615858167521375267910356975448388'
```

And in Py3:
```
>>> import binascii
# Note there's no trailing L here:
>>> var = 25923956239092482442895228561437790190304192615858167521375267910356975448388

# No trailing L on this string:
>>> hex(var)
'0x39507073526a42476a42683930497649533936646752635f554c364e79477144'

# Close, but we get bytes...
>>> binascii.unhexlify(hex(var)[2:])
b'9PpsRjBGjBh90IvIS96dgRc_UL6NyGqD'

# There we go, a string representing the encoded long:
>>> import six
>>> six.ensure_str(binascii.unhexlify(hex(var)[2:]))
'9PpsRjBGjBh90IvIS96dgRc_UL6NyGqD'
```

So, I think a Py2-and-3 compatible solution might look like this:

```
unhexed = binascii.unhexlify(hex(long_to_convert)[2:].rstrip('L'))
return six.ensure_str(unhexed)
```

Reply:
Woah, thank you for the info and a really nice solution!! I'll add that. 🙂 (also I think I forgot to finish editing this function and just left it in the debug print lol)